drivers/modem/modem_cellular.c

```diff
@@ -83,11 +84,16 @@ struct modem_cellular_data {
 	uint8_t *chat_argv[32];
 
 	/* Status */
-	uint8_t imei[15];
-	uint8_t hwinfo[64];
 	uint8_t registration_status_gsm;
 	uint8_t registration_status_gprs;
 	uint8_t registration_status_lte;
+	int8_t rssi;
+	uint8_t imei[15];
+	uint8_t model_id[64];
+	uint8_t imsi[15];
+	uint8_t iccid[22];
```

Comment:
Not so important, but doesn't ICCID have a maximum of 20 digits? It's just that for IMEI and IMSI we are a bit more accurate :-)

Reply:
I read otherwise https://www.wirelesslogic.com/iot-glossary/what-is-iccid/

> An ICCID is a unique 18 to 22-digit code used to identify a SIM (Subscriber Identity Module). You’ll sometimes see the ICCID referred to simply as the SIM Number.

Do you have a source ?

Reply:
I actually see that the country code is included in the ICCID, so this might go from one digit (+1 -> US), to two (+45 -> Denmark) or three (+345 -> Iceland), with the last case being 21 digits in total.

So, I take it back... let's stick to what you already had! :-)
I think my source were specific operators that weren't in, e.g. Iceland with three digits in the country code.